### PR TITLE
Lexi alternatives section

### LINKED ISSUE #33: alternative Phrases

### PR DESCRIPTION
**Pull request by LexiisYing**
## Description

- created alternatives button
- added css for the section
- reset functionality
- refactor code - code duplications have been minimised and logic has been simplified where possible
- created alternatives route (backend)

## Related to

Fixes #33

### CODE REVIEW

File: client/src/components/AlternativeButton.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+			const response = await fetch("http://localhost:3000/api/alternatives", {
+				method: "POST",
+				headers: { "Content-Type": "application/json" },
+				body: JSON.stringify({ input: outputValue }),
+			});
+			if (response.ok) {
+				const alternative = await response.json();
+				setAlternativeValue(alternative);
+			} else {
+				console.log("Error: " + response.status);
```
Comment (DomVinyard) [inline client/src/components/AlternativeButton.js:16]:
```suggestion
				throw new Error(response.status);
```

File: client/src/pages/Home.js [hunk 2]
```diff
@@ -67,12 +79,25 @@ const Home = () => {
...
+									value={alternativeValue}
 									readOnly
 								/>
+								<AlternativeButton
+									outputValue={outputValue}
+									setAlternativeValue={setAlternativeValue}
+								/>
+								<button onClick={handleReset} className="reset-button">
+									Reset
+								</button>
```
Comment (DomVinyard) [inline client/src/pages/Home.js:100]: duplication here?

File: server/api.js [hunk 3]
```diff
@@ -52,4 +52,47 @@ router.post("/", async (req, res) => {
...
+		});
+		const openai = new OpenAIApi(configuration);
+
+		const completion = await openai.createChatCompletion({
+			model: "gpt-3.5-turbo",
+			messages: [
+				{
+					role: "user",
+					content: `
+					You are a copywriter. Your job is to take some user text and make it better.
```
Comment (DomVinyard) [inline server/api.js:71]:
```suggestion
					You are a copywriter. Your job is to take some user text and offer alternative ways of phrasing it.
```

File: server/api.js [hunk 4]
```diff
@@ -52,4 +52,47 @@ router.post("/", async (req, res) => {
...
+					- Improve the grammar
+					- Make it more engaging 
+					- Make it as concise as possible without sacrificing clarity
```
Comment (DomVinyard) [inline server/api.js:74]:
```suggestion
```

File: client/src/components/AlternativeButton.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import "./AlternativeButton.css";
+
+const AlternativeButton = ({ outputValue, setAlternativeValue }) => {
+	const getAlternatives = async () => {
+		try {
+			const response = await fetch("http://localhost:3000/api/alternatives", {
```
Comment (Bahare09) [inline client/src/components/AlternativeButton.js:7]: @LexiisYing please change this line to `fetch("/api/alternatives", {`

**Review by Bahare09 — APPROVED** [review]
well done @LexiisYing

**Review by ZahraaTayyar — APPROVED** [review]
well done!

**Pull request merged**